Added Terminology and Workflow files for Git (#10)
helicalco/docs: content/learn/git/git-workflow.md

For instance:

```
*.log
*.exe
testfiles
```

In the above example, `git add .` will ignore any file or directory that ends with `.log` or `.exe` or is named `testfiles`.

### Commit Changes

Now to commit the changes you've just added to the staging branch.

```sh
$ git commit -m "first commit"
```

`git commit` will commit the changes you've made to the staging branch. You can also add a message to the commit using `-m` - which is highly recommended. All messages should be descriptive, and should be short - less than 50 characters. They should also begin with a lowercase letter.

### Linking to Github

Now you need to setup a remote repository. Head over to [Github](https://github.com/) and create a repository. To do this, click on the `new` button in the top right corner of the right hand side bar.
![new repository button](../images/github-new-repo-button.png)

It will then ask for a name and an optional description. You will be given a choice between public and private. Public means anyone who comes across your github page can see your repository. Private means only you can see your repository. You can change this later.

> Below that are some options to initialize the repository with. Don't worry about these for now. Since a repository is already created locally, it can create issues with the initial push if the remote repo isn't empty.

Reviewer: ```suggestion
Below that are some options to initialize the repository with. Leave the options unchecked, if set they can create conflicts between your local and remote repository.
```

![github repository options](../images/github-initialize-options.png)

### Link Remote and Push

> You have now successfully created a repository on github. Now you need to link tohe remote repository to your local repository. You will see some commands in your newly created empty repository on github. Take a look at the ones regarding pushing an existing repository.

Reviewer: ```suggestion
You have now successfully created a repository on github. Now you need to link the remote repository to your local repository. You will see some commands in your newly created empty repository on github. Take a look at the ones regarding pushing an existing repository.
```

Author: I miss spell/grammar check

Reviewer: I think there is a vscode extension for it, I had it at one point but it got annoying in actual code. Would be good to get but only run on `.md` files

Reviewer: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

![github connect local to remote](../images/github-connect-remote.png)

Lets work through these commands.

```sh
$ git remote add origin git@github.com:helical-tutorials/my-first-repository.git
```

> This will add a remote repository called `origin` to your local repository. This is the name of the remote repository. If you are to access anything branch directly on the remote, you prefix the branch name like `origin/<branch>` or `origin <branch>`. So this command links your local repository to the remote repository.

Reviewer: ```suggestion
This will add a remote repository called `origin` to your local repository. This is the name of the remote repository. If you are to access anything branch directly on the remote, prefix the branch name like `origin/<branch>` or `origin <branch>`. So this command links your local repository to the remote repository.
```

```sh
$ git branch -M main
```

> You can ignore this command if you've downloaded git recently. Github used to have the primary/default branch called master. This will change the name of the default branch from master to main.

Reviewer: ```suggestion
You can ignore this command if you've downloaded git recently. Github used to have the default branch called master. This will change the name of the default branch from master to main.
```

And there it is. You've made a remote repository and linked it to your local repository. You can now push your changes to the remote repository. Try making a change to one of your files and then push your changes to the remote repository. Here are the commands you will need:

```sh
$ git add .
$ git commit -m "made some changes"
$ git push
```

## Cloning a Repository

For this, you are going to clone your recently created repository from github into a new local repository.

Firstly, you need to create a new directory to clone the repository into. Then head over to github and select the repository you want to clone.

Click on the green code button, then the ssh tab. Copy the link provided.

![github clone link](../images/github-clone-code.png)

Using this link, head back to your terminal and within the empty directory you created, run the following command:

```sh
$ git clone git@github.com:helical-tutorials/my-first-repository.git .
```

> If you don't want to create a directory yourself, you can just run:

Reviewer: ```suggestion
If you don't want to create a directory yourself, you can instead run:
```
